settings.php

```diff
@@ -52,4 +53,19 @@
 
     $settings->add(new admin_setting_configcheckbox('questionnaire/allowemailreporting',
         get_string('configemailreporting', 'questionnaire'), get_string('configemailreportinglong', 'questionnaire'), 0));
+
+    // Manage old responses after. The default value is 24 months.
+    $options = questionnaire_create_remove_options();
+    $settings->add(new admin_setting_configselect('questionnaire/removeoldresponses',
+            get_string('removeoldresponsesafter', 'questionnaire'),
+            get_string('configremoveoldresponses', 'questionnaire'), 0, $options));
+
+    $options = [
```

Comment:
Ideally, this setting should appear before the previous setting ("questionnaire/removeoldresponses"), since it directly impacts that one.